aten/src/ATen/Context.h

```diff
@@ -214,22 +214,22 @@ static inline bool hasMKLDNN() {
 }
 
 static inline void manual_seed(uint64_t seed) {
-  auto& gen = globalContext().defaultGenerator(DeviceType::CPU);
+  auto gen = globalContext().defaultGenerator(DeviceType::CPU);
   {
     // See Note [Acquire lock when using random generators]
-    std::lock_guard<std::mutex> lock(gen.mutex_);
-    gen.set_current_seed(seed);
+    std::lock_guard<std::mutex> lock(gen->mutex_);
```

Comment:
Also doesn't have to be this PR, but we should consider hiding the mutex behind a method instead of exposing all the field guts to everyone.

Reply:
#34987